lib/rest-middleware.js

```diff
@@ -0,0 +1,77 @@
+function restMiddleware (connect, helmet, logger, restifyLinks) {
+    /**
+     * Sets up the self link of the current path we are on.
+     *
+     * @param {Object} req
+     * @param {Object} res
+     * @param {Function} next
+     */
+    function selfLink (req, res, next) {
+        res.links({
+            self: "https://" + req.header("host") + req.url
```

Comment:
We chatted about this coming from the `config.json` api.uri or some similar property.  Config should be set to a string that does not contain a trailing slash.